components/Repositories/ContributorConfidenceChart.tsx

```diff
@@ -0,0 +1,88 @@
+import { useMemo } from "react";
+import { FaUserPlus } from "react-icons/fa";
+import { PieChart, Pie, Cell, ResponsiveContainer, Label } from "recharts";
+import SkeletonWrapper from "components/atoms/SkeletonLoader/skeleton-wrapper";
+import Card from "components/atoms/Card/card";
+
+type ContributorConfidenceChartProps = {
+  contributorConfidence: number | undefined;
+  isLoading: boolean;
+  isError: boolean;
+  className?: string;
+};
+
+export default function ContributorConfidenceChart({
+  contributorConfidence,
+  isLoading,
+  isError,
+  className,
+}: ContributorConfidenceChartProps) {
+  const percentage = Math.floor((contributorConfidence ?? 0) * 100);
+  const data = useMemo(
+    () => [
+      { name: "confidence", value: percentage },
+      { name: "difference", value: 100 - percentage },
+    ],
+    [contributorConfidence]
+  );
+
+  const getValueBasedOnPercentage = ({ low, med, high }: { low: string; med: string; high: string }) => {
+    return percentage < 10 ? low : percentage < 30 ? med : high;
```

Comment:
I think these numbers should be lowers

low < 5%
med < 20%
high > 20%

But this could use some quick QA from @open-sauced/engineering

Reply:
+1 - I'd be shocked to find anything over 40% anywhere. That basically means that about ~1/2 of _all_ people staring / forking a repo come back to make a meaningful contribution. That'd be pretty insane since most people are just starring / forking stuff to play around or bookmark it for themselves.

One way we could swing this is to be based on 50% or 40% as the _max_. And anything over that is a total unicorn.

So:
- 40% should really be look like 80%
- 25% should look like 50%
- 10% should look like 20%
- etc.

---

I'd also recommend making the gauge react in a more meaningful way the difference between 7% and 40% should be astronomical:

![Screenshot 2024-06-10 at 5 47 00 PM](https://github.com/open-sauced/app/assets/23109390/44ca3136-22d8-47e1-b910-d37b2d8009f7)

![Screenshot 2024-06-10 at 5 47 06 PM](https://github.com/open-sauced/app/assets/23109390/ec3f95ca-975d-42ec-beb8-f818bd442f1a)

For 40%, maybe I'd expect to see an almost entirely full gauge:

![Screenshot 2024-06-10 at 5 48 51 PM](https://github.com/open-sauced/app/assets/23109390/0816d5bc-13ae-4c26-87ed-748022887821)

Reply:
I'd suggest not doing a full gauge but similar to an engine, anything over 40% should be red.

![image](https://github.com/open-sauced/app/assets/5713670/45937c78-b805-4432-80d5-6cfd8fe837b5)

Reply:
We should punt on this into a new issue/PR. I did find a consistent edge case:
- Docs: [dotnet/aspire-docs](https://deploy-preview-3545--oss-insights.netlify.app/s/dotnet/docs-aspire?range=90)
- Extensions: [rayacast](https://capture.dropbox.com/5PfURg4uQCwGprwi)

Reply:
I thought of suggesting winsorizing the data, but because the metric literally represents a % of contributors, it wouldn't make sense to change the actual number

What we can do is change the min/max on the gauge so that it reflects this better:

<img width="237" alt="image" src="https://github.com/open-sauced/app/assets/27322879/84fb8298-78e3-43fa-b36d-bfb5b0a024f9">


we could even reinforce the message with a low/med/high/very high badge and move the "Learn More" to the icon we've started using for explaining metrics.

<img width="641" alt="image" src="https://github.com/open-sauced/app/assets/27322879/2b314e82-23ce-4bc2-84f4-f903d25560c2">



@bdougie  I would highly suggest against adding red to the top limit though, it would make it look like it's something bad when it's not 😓

Reply:
> @bdougie I would highly suggest against adding red to the top limit though, it would make it look like it's something bad when it's not 😓

understood. 

> we could even reinforce the message with a low/med/high/very high badge and move the "Learn More" to the icon we've started using for explaining metrics.

Adding the high/mid/low badge would help. As well as adding a min/max. But I don't have strong feelings any of the above. I think if is a low effort way to make the current iteration more this is a subset (%) of contributors or provide an idea of what is the norm, that would help.

Today maintainers count stars, issues, and PRs which also doesn't scale.